specs/common/error-signalling.md

```diff
@@ -0,0 +1,155 @@
+# Error Signalling
+
+<!-- copied from ../serving/knative-api-specification-1.0.md#error-signalling -->
+
+Knative APIs use the
+[Kubernetes Conditions convention](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties)
+to communicate errors and problems to the user. Note that Knative customizes the
+general Kubernetes recommendation with a `severity` field, and does not include
+`lastTransitionTime` for scalability reasons. Each user-visible resource
+described in Resource Overview MUST have a `conditions` field in `status`, which
+MUST be a list of `Condition` objects of the following form. Fields in the
+condition which are not marked as REQUIRED may be omitted to indicate the
+default value (i.e. a Condition which does not include a `status` field is
+equivalent to a `status` of `"Unknown"`). The actual API object types in an
+OpenAPI document may be named `FooCondition` to allow better code generation and
+disambiguation between similar fields in the same `apiGroup`.
+
+<table>
+  <tr>
+   <td><strong>Field</strong>
+   </td>
+   <td><strong>Type</strong>
+   </td>
+   <td><strong>Description</strong>
+   </td>
+   <td><strong>Default Value If Unset</strong>
+   </td>
+  </tr>
+  <tr>
+   <td><code>type</code>
+   </td>
+   <td><code>string</code>
+   </td>
+   <td>The category of the condition, as a short, CamelCase word or phrase.
+<p>
+This is the primary key of the Conditions list when viewed as a map.
+   </td>
+   <td>REQUIRED – No default
+   </td>
+  </tr>
+  <tr>
+   <td><code>status</code>
+   </td>
+   <td>Enum:<ul>
+
+<li>"True"
+<li>"False"
+<li>"Unknown"</li></ul>
+
+   </td>
+   <td>The last measured status of this condition.
+   </td>
+   <td>"Unknown"
+   </td>
+  </tr>
+  <tr>
+   <td><code>reason</code>
+   </td>
+   <td>string
+   </td>
+   <td>One-word CamelCase reason for the condition's last transition.
+   </td>
+   <td>""
+   </td>
+  </tr>
+  <tr>
+   <td><code>message</code>
+   </td>
+   <td>string
+   </td>
+   <td>Human-readable sentence describing the last transition.
+   </td>
+   <td>""
+   </td>
+  </tr>
+  <tr>
+   <td><code>severity</code>
+   </td>
+   <td>Enum:<ul>
+
+<li>""
+<li>"Warning"
+<li>"Info"</li></ul>
+
+   </td>
+   <td>If present, represents the severity of the condition. An empty severity represents a severity level of "Error". 
```

Comment:
When I read the sentence below:
> Each resource MUST have either a Ready condition (for ongoing systems) or Succeeded condition (for resources that run to completion) with severity=""...

it reads kind of like there MUST be a severity=Error even when everything is ok. What am I missing?